src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertCarQuery = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection
+                        .prepareStatement(insertCarQuery, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert car: " + car, e);
+        }
+        if (car.getDrivers() != null && !car.getDrivers().isEmpty()) {
```

Comment:
це зайва перевірка, варто створювати car з проініціалізованим полем drivers, але просто з пустим
```suggestion
        if (!car.getDrivers().isEmpty()) {
```